pkg/eventlog/multiinterceptor.go

```diff
@@ -8,26 +8,23 @@ type repeater struct {
 	interceptors []Interceptor
 }
 
-func (r *repeater) InterceptWithReturn(events *events.EventList) (*events.EventList, error) {
-	panic("implement me")
-}
-
-func (r *repeater) Intercept(events *events.EventList) error {
+func (r *repeater) Intercept(events *events.EventList) (*events.EventList, error) {
 
 	// Avoid nil dereference if Intercept is called on a nil *Recorder and simply do nothing.
 	// This can happen if a pointer type to *Recorder is assigned to a variable with the interface type Interceptor.
 	// Mir would treat that variable as non-nil, thinking there is an interceptor, and call Intercept() on it.
 	// For more explanation, see https://mangatmodi.medium.com/go-check-nil-interface-the-right-way-d142776edef1
 	if r == nil {
-		return nil
+		return events, nil
 	}
 
 	for _, i := range r.interceptors {
-		if err := i.Intercept(events); err != nil {
-			return err
+		_, err := i.Intercept(events)
```

Comment:
In fact, contrary to what I wrote in an older comment, there is a totally meaningful way of defining the event modification semantics of the MultiInterceptor. We can simply feed the output of one interceptor to the input of the next one. This would correspond to a sort of "serial" connection of the interceptors and it would be the users responsibility to use a meaningful list of interceptors.
And the only code that needs to change to achieve that is this.

```suggestion
		events, err := i.Intercept(events)
```